pkg/output/output.go

```diff
@@ -241,26 +247,77 @@ func (w *StandardWriter) Close() error {
 
 // matchOutput checks if the event matches the output regex
 func (w *StandardWriter) matchOutput(event *Result) bool {
-	if w.matchRegex == nil {
+	if w.matchRegex == nil && w.outputMatchCondition == "" {
 		return true
 	}
 	for _, regex := range w.matchRegex {
 		if regex.MatchString(event.Request.URL) {
 			return true
 		}
 	}
-	return false
+
+	return evalDslExpr(event, w.outputMatchCondition)
 }
 
 // filterOutput returns true if the event should be filtered out
 func (w *StandardWriter) filterOutput(event *Result) bool {
-	if w.filterRegex == nil {
+	if w.filterRegex == nil && w.outputFilterCondition == "" {
 		return false
 	}
 	for _, regex := range w.filterRegex {
 		if regex.MatchString(event.Request.URL) {
 			return true
 		}
 	}
-	return false
+
+	return evalDslExpr(event, w.outputFilterCondition)
+}
+
+func evalDslExpr(result *Result, dslExpr string) bool {
+	resultMap, err := resultToMap(*result)
+	if err != nil {
+		gologger.Warning().Msgf("Could not map result: %s\n", err)
```

Comment:
I always think it's better to propagate upstream errors from packages:

```
func evalDslExpr(result *Result, dslExpr string) (bool, error) { ... }
```

but since we are already in the output package I think we can keep it this way.

These filtering conditions appears quite powerful also to discriminate which crawling branches abandoning or continue crawling. Shall we track to extend them for this functionality as well in the future? What do you think? (cc @ehsandeep )

Reply:
@Mzack9999 Yes, let's track it as a new issue. Can you please create an issue for it with minimal information?